### PR TITLE
rules: core: Add minValueInclusive rule

### PR DESCRIPTION
**Pull request by Lathrisk**
# #378

Adds a rule to check the values against minValueInclusive when it is supplied.

### CODE REVIEW

File: src/rules/core/minvalueinclusive-rule.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  constructor(options) {
+    super(options);
+    this.targetFields = '*';
+    this.meta = {
+      name: 'MinValueInclusiveRule',
+      description: 'Validates that all properties meet the associated minValueInclusive constraint.',
+      tests: {
+        belowMinimum: {
+          description: 'Raises a failure if the value is below the associated minValueInclusive property.',
+          message: 'The value of this property should be above or equal to {{minValueInclusive}}.',
```
Comment (nickevansuk) [inline src/rules/core/minvalueinclusive-rule.js:16]: Suggest slightly more conventional language here around "greater than".

```suggestion
          message: 'The value of this property must be greater than or equal to {{minValueInclusive}}.',
```

Also in general worth noting that within messages anything that results in a FAILURE must use the language "must", and anything that results in a WARNING must use the language "should". This is because "must" and "should" key words actually have defined interpretations (https://openactive.io/open-booking-api/EditorsDraft/1.0CR3/#conformance)

File: src/rules/core/minvalueinclusive-rule-spec.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    );
+    const errors = await rule.validate(nodeToTest);
+    expect(errors.length).toBe(0);
+  });
+
+  it('should return an error for a value below a minValueInclusive constraint', async () => {
+    const values = [
+      1,
+      -100.1,
+      -100,
```
Comment (nickevansuk) [inline src/rules/core/minvalueinclusive-rule-spec.js:65]: Is best practice to test boundary conditions, so e.g.
```suggestion
      -100.1,
      -100,
      -3.9,
```

File: src/rules/core/minvalueinclusive-rule-spec.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+  it('should target any field', () => {
+    const isTargeted = rule.isFieldTargeted(model, 'repeatCount');
+    expect(isTargeted).toBe(true);
+  });
+
+  it('should return no errors for a value above minValueInclusive constraint', async () => {
+    const values = [
+      89.12345,
+      89,
```
Comment (nickevansuk) [inline src/rules/core/minvalueinclusive-rule-spec.js:30]: Is best practice to test boundary conditions, so e.g.
```suggestion
      89.12345,
      89,
      4.1,
```

**Review by nickevansuk — APPROVED** [review]
Approved assuming minor changes

**Comment by nickevansuk** [comment]
@Lathrisk sorry just spotted this, should be:
> The value of this property **must** be greater than

anything that results in a FAILURE must use the language "must", and anything that results in a WARNING must use the language "should". This is because "must" and "should" key words actually have defined interpretations (https://openactive.io/open-booking-api/EditorsDraft/1.0CR3/#conformance)

**Comment by Lathrisk** [comment]
I'll put together an update shortly!
